cmd/otel-allocator/allocation/consistent_hashing.go

```diff
@@ -72,18 +73,18 @@ func newConsistentHashingAllocator(log logr.Logger) Allocator {
 // This method is called from within SetTargets and SetCollectors, which acquire the needed lock.
 // This is only called after the collectors are cleared or when a new target has been found in the tempTargetMap.
 // INVARIANT: c.collectors must have at least 1 collector set.
-func (c *consistentHashingAllocator) addTargetToTargetItems(target *TargetItem) {
+func (c *consistentHashingAllocator) addTargetToTargetItems(tg *target.Item) {
 	// Check if this is a reassignment, if so, decrement the previous collector's NumTargets
-	if previousColName, ok := c.collectors[target.CollectorName]; ok {
+	if previousColName, ok := c.collectors[tg.CollectorName]; ok {
 		previousColName.NumTargets--
 		TargetsPerCollector.WithLabelValues(previousColName.String(), consistentHashingStrategyName).Set(float64(c.collectors[previousColName.String()].NumTargets))
 	}
-	colOwner := c.consistentHasher.LocateKey([]byte(target.Hash()))
-	targetItem := &TargetItem{
-		JobName:       target.JobName,
-		Link:          LinkJSON{Link: fmt.Sprintf("/jobs/%s/targets", url.QueryEscape(target.JobName))},
-		TargetURL:     target.TargetURL,
-		Label:         target.Label,
+	colOwner := c.consistentHasher.LocateKey([]byte(tg.Hash()))
+	targetItem := &target.Item{
```

Comment:
nit: can we use a target constructor to make this `target.New`

Reply:
Thanks for the review! This is a good point, changed!